### DISCUSSION TITLE
Renovate bot does not update gradle dependencies declared in an array

### DISCUSSION THREAD

**Discussion by jreij** [renovatebot/renovate#28481] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We have a gradle array where we declare all our dependencies. Within this array we have multiple dependencies defined as a key-value structure and some others that are in an array form with values only (no keys). It seems that renovate is not updating those dependencies.

```groovy
ext {
    leak_canary_version = "2.13"
    appcompat_version = "1.6.1"
    // this is outdated
    coroutines_version = "1.6.4" 
    // this is also outdated
    lifecycle_version = "2.5.1"

    libraries = [
        leakCanary      : "com.squareup.leakcanary:leakcanary-android:$leak_canary_version",
        kotlinCoroutines: [
            // these dependencies do not get updated
            // they are grouped together in one array because we always need to import them together
            // we cannot add keys to them because the command to import them fails (check the last "implementation" line below)
            "org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version",
            "org.jetbrains.kotlinx:kotlinx-coroutines-android:$coroutines_version"
        ],
        androidx        : [
            appcompat       : "androidx.appcompat:appcompat:$appcompat_version",
            lifecycle       : [
                // these dependencies do not get updated
                "androidx.lifecycle:lifecycle-runtime-ktx:$lifecycle_version",
                "androidx.lifecycle:lifecycle-viewmodel-ktx:$lifecycle_version"
            ]
        ],
    ]
}


dependencies {
    implementation libraries.leakCanary
    implementation libraries.androidx.appcompat
    // this line imports the whole array directly
    implementation libraries.kotlinCoroutines
    // this line imports the whole array directly
    implementation libraries.androidx.lifecycle
}
```

I checked the logs of Mend renovate and I could not find any of these dependencies there, while I found the rest of the dependencies.

**Comment by rarkins** [comment]
Renovate doesn't support this custom approach to dependency definitions in Gradle

**Reply by Churro** [reply]
Correct, the non-matching is due to an array being used within a groovy map. I guess it would be feasible to add support for this, as the Kotlin `listOf` pendant is also supported. I'll put it on my list to check.

**Reply by jreij** [reply]
Thank you both! Looking forward to having this supported.

**Reply by rarkins** [reply]
@Churro do you need a reproduction repo?

**Reply by Churro** [reply]
Not necessarily but if @jreij provides one, I could use it in a PR for demonstration 👍

**Reply by Churro** [reply]
-> see PR https://github.com/renovatebot/renovate/pull/28517

**Reply by jreij** [reply, marked as answer]
I can verify that this is now fixed in [37.307.0](https://github.com/renovatebot/renovate/releases/tag/37.307.0). Thanks @Churro!
